code/android-core-library/src/main/java/com/adobe/marketing/mobile/internal/configuration/ConfigurationRulesManager.kt

```diff
@@ -69,25 +86,33 @@ internal class ConfigurationRulesManager(
     internal fun applyCachedRules(extensionApi: ExtensionApi): Boolean {
 
         if (configDataStore == null) {
-            Log.trace(ConfigurationExtension.TAG, LOG_TAG, "Cannot load rules from ${ConfigurationExtension.DATASTORE_KEY}. Cannot apply cached rules")
+            Log.trace(
+                ConfigurationExtension.TAG,
+                LOG_TAG,
+                "Cannot load rules from ${ConfigurationExtension.DATASTORE_KEY}. Cannot apply cached rules"
+            )
             return false
         }
 
         val persistedRulesUrl =
             configDataStore.getString(PERSISTED_RULES_URL, null)
 
         if (persistedRulesUrl.isNullOrBlank()) {
-            Log.trace(ConfigurationExtension.TAG, LOG_TAG, "Persisted rules url is null or empty. Cannot apply cached rules")
+            Log.trace(
+                ConfigurationExtension.TAG,
+                LOG_TAG,
+                "Persisted rules url is null or empty. Cannot apply cached rules"
+            )
             return false
         }
 
-        val cachedRulesDirectory: File? = cacheFileService
-            .getCacheFile(
-                persistedRulesUrl,
-                RULES_CACHE_FOLDER,
-                false
-            )
-        return replaceRules(cachedRulesDirectory, extensionApi)
+        val cachedRuleResult: CacheResult? = cacheService.get(RULES_CACHE_NAME, persistedRulesUrl)
```

Comment:
nit: Let us change this to rulesDownloader.loadFromCache(url) { ..} after adding the new URL.